store/store.go

```diff
@@ -0,0 +1,119 @@
+package store
+
+import (
+	"context"
+	"errors"
+	"sync"
+	"time"
+
+	"github.com/prometheus/alertmanager/types"
+	"github.com/prometheus/common/model"
+)
+
+type Store interface {
```

Comment:
That's a good point. I originally created it because I thought I would want to use some other struct that implemented the interface for testing, but there's really no need to mock this store. This pattern is also done in `provider` -- we only have the memory store, but it still is done as an interface.

I'm a bit conflicted. I 100% agree that since this is only being implemented by one struct that it doesn't need to be an interface, but there's something in the back of my mind that's saying designing for the interface might be a better idea. Not sure, will ruminate upon this. Designing by interface has been super helpful for me in the past, which was really the motivation here, but that doesn't mean we need to stick with having an actual `interface`.